test/denormalized_has_many_test.rb

```diff
@@ -12,7 +12,23 @@ def setup
     @record.reload
   end
 
+  def test_uncached_record_from_the_db_will_use_normal_association_as_ary
+    # Test configuration where fetch association returns an array
+    IdentityCache.allow_fetch_to_return_array do    
+      assert_equal IdentityCache.fetch_returns_relation, false
+      expected = @record.associated_records.to_a
+      record_from_db = Item.find(@record.id)
+      assert_queries(1) do
+        assert_memcache_operations(0) do
```

Comment:
I could surely focus on the class of the fetch result.

I had the number of queries checked to make sure we had the right behaviour for both cached & uncached records; and ensure tests in the previous [PR](https://github.com/Shopify/identity_cache/pull/224/files) were still passing.